src/main/webapp/app/app.component.html

```diff
@@ -9,7 +9,9 @@
   </mat-toolbar>
   <section class="content">
     <div class="wrapper">
-      <tiny-task-list [tasks]="tasks$ | async" (deleted)="deleted($event)"></tiny-task-list>
+      <tiny-search [(value)]="term"></tiny-search>
```

Comment:
what made you chose this binding over using an eventEmitter here like e.g. (search)?

Reply:
At the moment, my changes are specifically for the search feature.
Later, in case having more time, my plan was to transform the search to a global component form-input-text.
The objective behind it is to encapsulate the mat-form-field in one single component to be used by the other ones. 
Please check the sample below:

```
// shared/form-input-text/form-input-text.ts

import { Component, forwardRef, Input } from '@angular/core';
import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';

/**
 * Displays regular text input.
 */
@Component({
    selector: 'tiny-form-input-text',
    templateUrl: './form-input-text.component.html',
    styleUrls: ['./form-input-text.component.scss'],
    providers: [
        {
            provide: NG_VALUE_ACCESSOR,
            useExisting: forwardRef(() => FormInputTextComponent),
            multi: true,
        },
    ],
})
export class FormInputTextComponent implements ControlValueAccessor {
    @Input() placeholder = '';
    @Input() label = '';
    @Input() icon = '';
    @Input() error = '';
    isTouched = false;
    value = '';

    registerOnChange(onChange: (value: string) => void): void {
        this.onChange = onChange;
    }

    registerOnTouched(onTouched: () => void): void {
        this.onTouched = onTouched;
    }

    writeValue(value: string): void {
        this.value = value || '';
    }

    pushChanges(value: string) {
        this.writeValue(value);
        this.onChange(this.value);
    }

    pushTouches() {
        this.isTouched = true;
        this.onTouched();
    }

    protected onTouched = () => {};
    protected onChange = (value: string) => {};
}
```